checkout-core/src/main/java/com/adyen/checkout/core/model/ModelObject.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2020 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by caiof on 17/12/2020.
+ */
+package com.adyen.checkout.core.model
+
+import android.os.Parcel
+import android.os.Parcelable
+import com.adyen.checkout.core.exception.CheckoutException
+import com.adyen.checkout.core.model.JsonUtils.readFromParcel
+import com.adyen.checkout.core.model.ModelObject.Serializer
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Base class for a Model object. A model object is a representation of a JSON response or input from the Checkout API.
+ * All model objects can be serialized and deserialized to and from a JSONObject using the [Serializer] interface.
+ * All model object also implement Parcelable to be sent as part of Extras in an Intent Bundle.
+ *
+ *
+ *
+ * The classes extending [ModelObject] are data classes designed to work standalone or in association with JSON libraries like GSON and Moshi.
+ */
+abstract class ModelObject : Parcelable {
+    override fun describeContents(): Int {
+        return Parcelable.CONTENTS_FILE_DESCRIPTOR
+    }
+
+    /**
+     * Interface that must be implemented and provided as a public SERIALIZER field that serializes the to and from a JSONObject.
+     * @param <T> The class that extends ModelObject to be serialized.
+     */
+    interface Serializer<T : ModelObject?> {
```

Comment:
Is the nullability intended?

Reply:
No it wasn't, I was already seeing issue in the next refactors, good catch.